src/atlas/projection/detail/StretchRegular.cc

```diff
@@ -0,0 +1,385 @@
+*
+* The origin of the xy-system is at (lon,lat) = (0,0)
+*
+*/
+
+namespace atlas {
+namespace projection {
+namespace detail {
+
+
+///< specification parameters
+template <typename Rotation>
+typename StretchLAM<Rotation>::Spec StretchLAM<Rotation>::spec() const {
+
+    Spec proj_st;
+    proj_st.set( "delta_low", delta_low_ ); ///< resolution of the external regular grid (rim)
+    proj_st.set( "delta_high", delta_high_ ); ///< resolution of the regional model (regular grid)
+    proj_st.set( "var_ratio", var_ratio_ ); ///< power used for the stretching
+    proj_st.set( "x_reg_start", x_reg_start_ ); ///< xstart of the internal regional grid
+    proj_st.set( "x_reg_end", x_reg_end_ ); ///< xend of the internal regional grid
+    proj_st.set( "y_reg_start", y_reg_start_); ///< ystart of the internal regional grid
+    proj_st.set( "y_reg_end", y_reg_end_ ); ///< yend of the internal regional grid
+    proj_st.set( "startx", startx_ ); ///< original domain startx
+    proj_st.set( "endx", endx_ ); ///< original domain endx
+    proj_st.set( "starty", starty_ ); ///< original domain starty
+    proj_st.set( "endy", endy_ ); ///< original domain endy
+    proj_st.set( "rim_widthx", rim_widthx_ ); ///< xsize of the rim
+    proj_st.set( "rim_widthy", rim_widthy_ ); ///< ysize of the rim
+
+    return proj_st;
+}
+
+///< constructors
+template <typename Rotation>
+StretchLAM<Rotation>::StretchLAM(const eckit::Parametrisation& proj_st) :
+    ProjectionImpl(), rotation_( proj_st ) {
+
+
+    proj_st.get( "delta_low", delta_low_ = 0.); ///< resolution of the external regular grid (rim)
+    proj_st.get( "delta_hi", delta_high_ = 0. ); ///< resolution of the regional model (regular grid)
+    proj_st.get( "var_ratio", var_ratio_ = 0. ); ///< power used for the stretching
+    proj_st.get( "x_reg_start", x_reg_start_ = 0. ); ///< xstart of the internal regional grid
+    proj_st.get( "y_reg_start", y_reg_start_ = 0.); ///< ystart of the internal regional grid
+    proj_st.get( "x_reg_end", x_reg_end_ = 0. ); ///< xend of the regular part of stretched internal grid
+    proj_st.get( "y_reg_end", y_reg_end_ = 0.); ///< yend of the regular part of stretched internal grid
+    proj_st.get( "startx", startx_ = 0. ); ///< original domain startx
+    proj_st.get( "endx", endx_ = 0. ); ///< original domain endx
+    proj_st.get( "starty", starty_ = 0. ); ///< original domain starty
+    proj_st.get( "endy", endy_ = 0. ); ///< original domain endy
+    proj_st.get( "rim_widthx", rim_widthx_ ); ///< xsize of the rim
+    proj_st.get( "rim_widthy", rim_widthy_ ); ///< ysize of the rim
+}
+
+template<typename Rotation>
+void StretchLAM<Rotation>::checkvalue(const double & epsilon,
+                                        const double & value_check) const {
+    std::string err_message;
+    std::string str = std::to_string(value_check);
+    err_message = "USER defined limits not in the middle of the area " + str;
```

Comment:
These strings are created regardless of the error. You will see a big speedup if these are instead within the following `if` statement.

Reply:
Done.

Reply:
Thanks, I don't think it is pushed yet. Note that I added a few commits to the branch.